environs/networking.go

```diff
@@ -105,3 +117,19 @@ func SupportsContainerAddresses(env Environ) bool {
 	}
 	return ok
 }
+
+// ProviderSpaceInfo contains all the information about a space needed
+// by another environ to decide whether it can be routed to.
+type ProviderSpaceInfo struct {
+	network.SpaceInfo
+
+	// This identifies the cloud containing the space. Care must be
```

Comment:
I'm still not convinced that we need *any* structure around the information. This is better than the layer above passing it in, but some of the bits in CloudSpec are never going to be relevant: IdentityEndpoint, StorageEndpoint.

I guess my only question is: do we think we'll ever want to use the CloudSpec field *outside* the provider? Maybe show which cloud/region a remote model is in to the user? Then some structure would be useful. But within the provider, the provider creates the information so it can apply its own structure to it.